tuf/data/types.go

```diff
@@ -210,23 +210,18 @@ func (f FileMeta) Equals(o FileMeta) bool {
 	return bytes.Equal(fBytes, oBytes)
 }
 
-// CheckHashes verifies all the checksums specified by the "hashes" of the payload.
-func CheckHashes(payload []byte, name string, hashes Hashes) error {
+func checkHashes(s *StreamingHasher, name string, hashes Hashes) error {
 	cnt := 0
 
 	// k, v indicate the hash algorithm and the corresponding value
 	for k, v := range hashes {
 		switch k {
-		case notary.SHA256:
-			checksum := sha256.Sum256(payload)
-			if subtle.ConstantTimeCompare(checksum[:], v) == 0 {
-				return ErrMismatchedChecksum{alg: notary.SHA256, name: name, expected: hex.EncodeToString(v)}
-			}
-			cnt++
-		case notary.SHA512:
-			checksum := sha512.Sum512(payload)
-			if subtle.ConstantTimeCompare(checksum[:], v) == 0 {
-				return ErrMismatchedChecksum{alg: notary.SHA512, name: name, expected: hex.EncodeToString(v)}
+		case notary.SHA256, notary.SHA512:
+			// skipping the boolean check, because all internal usages ensure that the correct
+			// hash algorithms are produced
+			checksum, _ := s.GetChecksum(k)
```

Comment:
this works, but I'm wondering if we should be more defensive here and skip the constant time compare if  `s.GetChecksum` returns false in the second return value?